8802: sample publish/recall audit events

labkey-kevink opened the pull request:
#### Rationale
Create an aggregate publish event when linking a group of SampleType rows to a study.  Add sample timeline events for each SampleType row as well.  Similarly, add recall events when removing SampleType rows from a study dataset.

#### Related Pull Requests
* https://github.com/LabKey/provenance/pull/67
#### Changes
* Rename AssayAuditProvider to PublishAuditProvider and support SampleType publish/recall events
* Conditionally add the "Link to Study" and "Recall" buttons only if study is available
* Teach the publish history views to support SampleType events
* Create provenance run linking the published SampleType -> dataset rows

labkey-kevink committed:
audit log SampleType link-to-study and recall-from-study events

labkey-kevink committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

# Conflicts:
#	experiment/src/org/labkey/experiment/api/ExperimentServiceImpl.java
#	study/src/org/labkey/study/assay/StudyPublishManager.java
#	study/src/org/labkey/study/assay/query/AssayAuditProvider.java

labkey-kevink committed:
clean up javadoc

labkey-kevink committed:
key(Enum)

labkey-kevink committed:
use "linked to study" in audit log message

labkey-kevink committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

labkey-kevink committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

labkey-kevink committed:
- wire up "link to study" button for SampleType if study module is available
- move SampleType study publish actions from experiment into study module
- move publish jsp next to their actions
- log SampleType publish and recall audit events in PublishAuditProvider and SampleTimelineAuditProvider

labkey-kevink committed:
- add PublishHistoryDetailsForm and fix parameters
- include assay name or sample type name (sourceName) in publish audit log

labkey-klum committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

labkey-klum commented on api/src/org/labkey/api/exp/api/ExperimentUrls.java:
```diff
@@ -32,6 +33,11 @@
+    {
+        return PageFlowUtil.urlProvider(ExperimentUrls.class);
+    }
+
```
nice

labkey-klum reviewed (approved):
I saw you added code to create a provenance run for samples. Are we cleaning up that run if the source gets deleted or is that something that needs to be handled later?

labkey-klum commented:
@labkey-susanh if you are interested in reviewing how publish/recall events are incorporated into `SampleTimeline` take a look at the changes in `StudyPublishManager`

labkey-kevink committed:
- delete provenance rows when a sample is deleted
- refactor provenance delete to handle dataset rows, assay result rows, and sample type rows

labkey-kevink commented:
> I saw you added code to create a provenance run for samples. Are we cleaning up that run if the source gets deleted or is that something that needs to be handled later?

Good catch.  We were cleaning up the provenance data when the dataset rows were recalled, but not when the sample rows were deleted.  We will now cleanup the provenance data and StudyPublish run if the source samples are deleted.  I've refactored the provenance delete code to handle dataset, assay result, and sample LSIDs.

labkey-kevink committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

labkey-kevink committed:
Merge branch 'develop' into fb_sample_publish_audit_8802

labkey-kevink committed:
use audit event's container column for SampleTypeID URL container context
